qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -150,10 +151,11 @@ def __init__(
         self.skip_routing = skip_routing
         if self.coupling_map is not None:
             if not self.coupling_map.is_symmetric:
-                # deepcopy is needed here to avoid modifications updating
-                # shared references in passes which require directional
-                # constraints
-                self.coupling_map = copy.deepcopy(self.coupling_map)
+                # deepcopy is needed here if we don't own the coupling map (i.e. we were passed it
+                # directly) to avoid modifications updating shared references in passes which
+                # require directional constraints
+                if isinstance(coupling_map, CouplingMap):
```

Comment:
This confused me a bit at first until I realized that the this wasn't `self.coupling` it was the input argument. I think this works fine, but it might be better to use `self.target is None` like in `SabreSwap`

Reply:
I actually thought this was a bit clearer, since it's a direct test of what we need to know.  I'm happy to normalise both to whichever you prefer, though.

Reply:
Changed SabreSwap in 3f1f493aa.